homeassistant/components/tuya/__init__.py

```diff
@@ -0,0 +1,296 @@
+    TUYA_DEVICE_MANAGER,
+    TUYA_DISCOVERY_NEW,
+    TUYA_HA_DEVICES,
+    TUYA_HA_TUYA_MAP,
+    TUYA_HOME_MANAGER,
+    TUYA_MQTT_LISTENER,
+    TUYA_SETUP_PLATFORM,
+    TUYA_SUPPORT_HA_TYPE,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    vol.All(
+        cv.deprecated(DOMAIN),
+        {
+            DOMAIN: vol.Schema(
+                {
+                    vol.Required(CONF_PROJECT_TYPE): int,
+                    vol.Required(CONF_ENDPOINT): cv.string,
+                    vol.Required(CONF_ACCESS_ID): cv.string,
+                    vol.Required(CONF_ACCESS_SECRET): cv.string,
+                    CONF_USERNAME: cv.string,
+                    CONF_PASSWORD: cv.string,
+                    CONF_COUNTRY_CODE: cv.string,
+                    CONF_APP_TYPE: cv.string,
+                }
+            )
+        },
+    ),
+    extra=vol.ALLOW_EXTRA,
+)
+
+# decrypt or encrypt entry info
+
+
+def entry_decrypt(hass: HomeAssistant, entry: ConfigEntry, init_entry_data):
+    """Decrypt code from config entry."""
+    aes = Aes()
+    # decrypt the new account info
+    if XOR_KEY in init_entry_data:
+        _LOGGER.debug("tuya.__init__.exist_xor_cache-->True")
+        key_iv = aes.xor_decrypt(init_entry_data[XOR_KEY], init_entry_data[KEY_KEY])
+        cbc_key = key_iv[0:16]
+        cbc_iv = key_iv[16:32]
+        decrpyt_str = aes.cbc_decrypt(cbc_key, cbc_iv, init_entry_data[AES_ACCOUNT_KEY])
+        entry_data = aes.json_to_dict(decrpyt_str)
+    else:
+        # if not exist xor cache, use old account info
+        _LOGGER.debug("tuya.__init__.exist_xor_cache-->False")
+        entry_data = init_entry_data
+        cbc_key = aes.random_16()
+        cbc_iv = aes.random_16()
+        access_id = init_entry_data[CONF_ACCESS_ID]
+        access_id_entry = aes.cbc_encrypt(cbc_key, cbc_iv, access_id)
+        merged_data = cbc_key + cbc_iv
+        c_xor_entry = aes.xor_encrypt(merged_data, access_id_entry)
+        # account info encrypted with AES-CBC
+        user_input_encrpt = aes.cbc_encrypt(
+            cbc_key, cbc_iv, json.dumps(dict(init_entry_data))
+        )
+        # update old account info
+        hass.config_entries.async_update_entry(
+            entry,
+            data={
+                AES_ACCOUNT_KEY: user_input_encrpt,
+                XOR_KEY: c_xor_entry,
+                KEY_KEY: access_id_entry,
+            },
+        )
+    return entry_data
+
+
+async def _init_tuya_sdk(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    init_entry_data = entry.data
+    # decrypt or encrypt entry info
+    entry_data = entry_decrypt(hass, entry, init_entry_data)
+    project_type = ProjectType(entry_data[CONF_PROJECT_TYPE])
+    api = TuyaOpenAPI(
+        entry_data[CONF_ENDPOINT],
+        entry_data[CONF_ACCESS_ID],
+        entry_data[CONF_ACCESS_SECRET],
+        project_type,
+    )
+
+    api.set_dev_channel("hass")
+
+    if project_type == ProjectType.INDUSTY_SOLUTIONS:
```

Comment:
Typo in the project name. This will need fixing upstream.

```suggestion
    if project_type == ProjectType.INDUSTRY_SOLUTIONS:
```

Reply:
fixed

Reply:
This is not fixed yet.